### PR TITLE
FOUNDATIONS: Retrieve All ShoppingItems

### PR DESCRIPTION
**Pull request by tharaniannamalai**

### CODE REVIEW

File: ShoppingList.Tests.Unit/Services/Foundations/ShoppingItems/ShoppingItemServiceTests.Exceptions.RetrieveAll.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            var failedShoppingItemServiceException =
+                new FailedShoppingItemServiceException(serviceException);
+
+            var expectedShoppingItemServiceException =
+                new ShoppingItemServiceException(failedShoppingItemServiceException);
+
+            this.storageBrokerMock.Setup(broker =>
+                broker.SelectAllShoppingItems())
+                    .Throws(serviceException);
+            // when
```
Comment (ShriHumrudha) [inline ShoppingList.Tests.Unit/Services/Foundations/ShoppingItems/ShoppingItemServiceTests.Exceptions.RetrieveAll.cs:30]: New line break required

File: ShoppingList.ConsoleApp/Services/Foundations/ShoppingItems/ShoppingItemService.cs [hunk 2]
```diff
@@ -20,13 +21,19 @@ public partial class ShoppingItemService : IShoppingItemService
...
         {
             ValidateShoppingItem(shoppingItem);
 
             return this.storageBroker.InsertShoppingItem(shoppingItem);
         });
+
+        public List<ShoppingItem> RetrieveAllShoppingItems() =>
+        TryCatch(() =>
+        {
+           return this.storageBroker.SelectAllShoppingItems();
```
Comment (ShriHumrudha) [inline ShoppingList.ConsoleApp/Services/Foundations/ShoppingItems/ShoppingItemService.cs]: Correct the tab alignment here

File: ShoppingList.Tests.Unit/Services/Foundations/ShoppingItems/ShoppingItemServiceTests.cs [hunk 3]
```diff
@@ -38,7 +40,12 @@ public ShoppingItemServiceTests()
                && (actualException.InnerException as Xeption).DataEquals(expectedException.InnerException.Data);
         }
 
-        private ShoppingItem CreateRandomShoppingItem() =>
+        private static int GetRandomNumber() =>
+            new IntRange(min: 2, max: 10).GetValue();
+        private static List<ShoppingItem> CreateRandomShoppingItems() =>
```
Comment (ShriHumrudha) [inline ShoppingList.Tests.Unit/Services/Foundations/ShoppingItems/ShoppingItemServiceTests.cs:46]: Need new line break

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]

File: ShoppingList.Tests.Unit/Services/Foundations/ShoppingItems/ShoppingItemServiceTests.Exceptions.RetrieveAll.cs (hunk 1 above)
Comment (tharaniannamalai) [reply ShoppingList.Tests.Unit/Services/Foundations/ShoppingItems/ShoppingItemServiceTests.Exceptions.RetrieveAll.cs:30]: Comment Resolved.

File: ShoppingList.ConsoleApp/Services/Foundations/ShoppingItems/ShoppingItemService.cs (hunk 2 above)
Comment (tharaniannamalai) [reply ShoppingList.ConsoleApp/Services/Foundations/ShoppingItems/ShoppingItemService.cs]: Comment Resolved.

File: ShoppingList.Tests.Unit/Services/Foundations/ShoppingItems/ShoppingItemServiceTests.cs (hunk 3 above)
Comment (tharaniannamalai) [reply ShoppingList.Tests.Unit/Services/Foundations/ShoppingItems/ShoppingItemServiceTests.cs:46]: Comment Resolved.

**Review by ShriHumrudha — APPROVED** [review]

**Pull request merged**